icu4c/source/i18n/number_capi.cpp

```diff
@@ -225,6 +251,144 @@ unumf_close(UNumberFormatter* f) {
 }
 
 
+///// SIMPLE NUMBER FORMATTER /////
+
+U_CAPI USimpleNumber* U_EXPORT2
+usnum_openForInt64(int64_t value, UErrorCode* ec) {
+    auto* impl = new USimpleNumberData();
+    if (impl == nullptr) {
+        *ec = U_MEMORY_ALLOCATION_ERROR;
+        return nullptr;
+    }
+    impl->fNumber = SimpleNumber::forInt64(value, *ec);
+    return impl->exportForC();
+}
+
+U_CAPI void U_EXPORT2
+usnum_multiplyByPowerOfTen(USimpleNumber* unumber, int32_t power, UErrorCode* ec) {
+    auto* number = USimpleNumberData::validate(unumber, *ec);
```

Comment:
just FYI:
most C APIs don't validate the pointer type via magic-number bytes; they just cast the pointer
- https://github.com/unicode-org/icu/blob/main/icu4c/source/common/normalizer2.cpp#L346
- https://github.com/unicode-org/icu/blob/main/icu4c/source/common/ubrk.cpp#L227
- https://github.com/unicode-org/icu/blob/main/icu4c/source/i18n/ucol.cpp#L461

Reply:
acknowledged. I wrote all of the NumberFormatter C APIs to do this with @aheninger so I'm just following the pattern in this file.